mne_bids/copyfiles.py

```diff
@@ -339,6 +341,105 @@ def copyfile_brainvision(vhdr_src, vhdr_dest, anonymize=None, verbose=False):
+    dest : str
+        The destination path of the .edf or .bdf file.
+    anonymize : dict | None
+        If None (default), no anonymization is performed.
+        If dict, data will be anonymized depending on the keys provided with
+        the dict: `daysback` is a required key, `keep_his` is an optional key.
+
+        `daysback` : int
+            Number of days by which to move back the recording date in time.
+            In studies with multiple subjects the relative recording date
+            differences between subjects can be kept by using the same number
+            of `daysback` for all subject anonymizations. `daysback` should be
+            great enough to shift the date prior to 1925 to conform with BIDS
+            anonymization rules. Due to limitations of the EDF/BDF format, the
+            year of the anonymized date will always be set to 1985 in the
+            'meas_date' region of the file. The correctly-shifted year will be
+            written to the 'recording info' region of the header, which may not
+            be parsed by all EDF/BDF reader software.
+
+        `keep_his` : bool
+            By default (False), all subject information next to the recording
+            date will be overwritten as well. If True, keep subject information
+            apart from the recording date. Participant names and birthdates
+            will always be anonymized if present, regardless of this setting.
+
+    See Also
+    --------
+    mne.io.anonymize_info
+
+    """
+    # Ensure source & destination extensions are the same
+    fname_src, ext_src = _parse_ext(src)
+    fname_dest, ext_dest = _parse_ext(dest)
+    if ext_src != ext_dest:
+        raise ValueError(f'Need to move data with same extension'
+                         f' but got "{ext_src}", "{ext_dest}"')
+
+    # Copy data prior to any anonymization
+    sh.copyfile(src, dest)
+
+    # Anonymize EDF/BDF data, if requested
+    if anonymize is not None:
+        if ext_src == '.bdf':
+            raw = read_raw_bdf(dest, preload=False, verbose=0)
+        else:
+            raw = read_raw_edf(dest, preload=False, verbose=0)
+
+        # Get patient info, recording info, and recording date
+        with open(dest, 'rb') as f:
+            f.seek(8)  # id_info field starts 8 bytes in
+            id_info = f.read(80).decode('ascii').rstrip()
+            rec_info = f.read(80).decode('ascii').rstrip()
+
+        # Parse metadata from file
+        if len(id_info) == 0 or len(id_info.split(' ')) != 4:
+            id_info = "X X X X"
+        if len(rec_info) == 0 or len(rec_info.split(' ')) != 5:
+            rec_info = "Startdate X X X X"
+        pid, sex, birthdate, name = id_info.split(' ')
+        startdate, admin_code, tech, equip = rec_info.split(' ')[1:5]
+
+        # Anonymize recording date, using 4-digit year from recording info
+        # if possible (regular EDF year only stores last two digits)
```

Comment:
```suggestion
        # Try to anonymize recording date using 4-digit year from
        # "Startdate" field from "recording info" section, because
        # The standard "startdate" field only supports 2-digit years.
        #
        # XXX: recording dates are specified twice in EDF/EDF+/BDF,
        # MNE-Python currently parses the 2-digit year field. If that is
        # changed to defaulting to the 4-digit field, this try-except
        # clause can be removed. See:
        # https://github.com/mne-tools/mne-python/issues/8544
        
```